### PR TITLE
setup graphql codegen

### PR DESCRIPTION
**Pull request by elbotho**
some wips:

- ~create-title is also used in revision and subscriptions~ done
- ~dedupe dependencies~ done
- ~breadcrumbs is used to fetch editor data~ marked as todo
- ~revision query not finished yet~ done
- ~manage subscriptions with custom title finder~ replacement done
- test if everything is fine

### CODE REVIEW

**Comment by Entkenntnis** [comment]
@elbotho thank your for the preparing work and sorry for the late review!

I would really like to use codegen to autogenerate query types and graphql codegen looks like a great candidate :)

I'm happy to help with integrating this, I think the biggest challenges will be the refactoring of our code to use the generated types, this could take some time (but I'm glad if we can delete all manually written query codes)

**Comment by elbotho** [comment]
> I'm happy to help with integrating this, 

Nice! 🎉

> I think the biggest challenges will be the refactoring of our code to use the generated types, this could take some time 

That's true. Also we will have to fix some typescript errors in the process for sure.
But we should be able to do this iteratively query by query.

**Comment by Entkenntnis** [comment]
I'll try to start with the main query. If this is successful, then we can merge this and continue to work iteratively.

**Comment by Entkenntnis** [comment]
Main query type checks :100:, but it's still messy

some thoughts:

- dev experience is really nice! I can change the raw query, run `yarn codegen`, and the types are magically correct
- some typescript magic like `Omit` or `Extract` is needed to get correct types
- It reveals some subtle bugs and inconsistencies we are having in our `create-XXX` functions -> a good time to review all this.

todo:

- [x] review this PR slowly and check if everything is fine
- [x] fix test

File: package.json [hunk 1]
```diff
@@ -9,7 +9,7 @@
   "license": "Apache-2.0",
   "author": "Serlo Education e.V.",
   "scripts": {
-    "_eslint": "eslint \"src/**/*.{js,jsx,ts,tsx}\"",
+    "_eslint": "eslint \"src/edtr-io/**/*.{js,jsx,ts,tsx}\"",
```
Comment (elbotho) [inline package.json]: on purpose?

File: src/components/pages/manage-subscriptions.tsx [hunk 2]
```diff
@@ -90,3 +92,30 @@ export function ManageSubscriptions({
...
+    object.__typename == 'Applet' ||
+    object.__typename == 'CoursePage' ||
+    object.__typename == 'Course' ||
+    object.__typename == 'Event'
+  ) {
+    return object.currentRevision?.title
+  }
+
+  return null
+}
```
Comment (elbotho) [inline src/components/pages/manage-subscriptions.tsx]: this whole block is more or less duplicated now, right?

**Comment by elbotho** [comment]
I renamed `MainPageQuery` to `MainUuidQuery` and tried to simplify some types. 

Other than the commented: LGTM.
Thank you for proceeding with this.

File: package.json (hunk 1 above)
Comment (Entkenntnis) [reply package.json]: reverted

File: src/components/pages/manage-subscriptions.tsx (hunk 2 above)
Comment (Entkenntnis) [reply src/components/pages/manage-subscriptions.tsx]: I have recombined the queries (good an idea how to do this :)

**Pull request merged**